pkg/cloud/awsprovider.go

```diff
@@ -2314,3 +2315,15 @@ func (aws *AWS) PricingSourceSummary() interface{} {
 	// encode the pricing source summary as a JSON string
 	return aws.Pricing
 }
+
+func (aws *AWS) GetNodePoolName(labels map[string]string) string {
+	sanitizedLabel := regexp.MustCompile(`[^a-zA-Z0-9 ]+`).ReplaceAllString(EKSNodepoolLabel, "_")
+
+	if poolName, found := labels[fmt.Sprintf("label_%s", sanitizedLabel)]; found {
+		return poolName
+	} else {
+		log.Warnf("unable to derive node pool name from node labels")
+		return ""
```

Comment:
This is more of a curiosity, but why return an empty string instead of nil?

Reply:
In go, you actually cannot have nil strings, unless you have a string pointer. So in stuff like structs, when you instantiate a new one, all of your string members are set to ""

Reply:
Gotcha! I figured it was something like that. I'm still learning Go, so I appreciate you sharing that info.